### PR TITLE
Remove some modules

### PR DESCRIPTION
**Pull request by blampe**
This removes three unnecessary modules:
* pf/go.mod
* x/muxer/go.mod
* testing/go.mod

And three unnecessary test modules:
* pf/tests/go.mod
* pkg/tests/go.mod
* x/muxer/tests/go.mod

This is intended to be merged with #2477 and #2480, which combine to make this a no-op for existing consumers.

Reviewers please double check that testing still works as expected locally as well as on CI. AFAICT `make test` is intended to run everything, and it had to take extra steps to account for all the modules, but a `go test ./...` at the root now captures _most_ (not all) of our tests.

Fixes https://github.com/pulumi/pulumi-terraform-bridge/issues/1445.

### CODE REVIEW

**Review by iwahbe — CHANGES_REQUESTED** [review]
Automated upgrades won't work for all providers that use PF. 

At minimum, we will need a roll-out plan for upgrading each provider?

If we do this, should we cut a release that deprecates `/pf` to tell users to switch from `/pf` to `/v3/pf`?

**Comment by blampe** [comment]
> Automated upgrades won't work for all providers that use PF.
> 
> At minimum, we will need a roll-out plan for upgrading each provider?

We can modify the upgrade automation to apply this patch:
```diff
@@
var alias identifier
@@
-import alias "github.com/pulumi/pulumi-terraform-bridge/pf"
+import alias "github.com/pulumi/pulumi-terraform-bridge/v3/pf"
@@
var alias identifier
@@
-import alias "github.com/pulumi/pulumi-terraform-bridge/x/muxer"
+import alias "github.com/pulumi/pulumi-terraform-bridge/v3/x/muxer"
```

(Or whatever the new import path ends up being.)

> If we do this, should we cut a release that deprecates `/pf` to tell users to switch from `/pf` to `/v3/pf`?

I think that's essentially what @VenelinMartinov was trying to do in #2358 -- leave the old module hanging around, but provide a new import to use instead. If we can do that it would be a smoother transition, but I don't know how feasible it is. Maybe since this PR is passing it will help connect some of the dots.

**Comment by blampe** [comment]
> Additional details and impacted files
> ```diff
> @@            Coverage Diff             @@
> ##           master    #2473      +/-   ##
> ==========================================
> + Coverage   57.19%   61.77%   +4.57%     
> ==========================================
>   Files         372      380       +8     
>   Lines       51243    51263      +20     
> ==========================================
> + Hits        29307    31666    +2359     
> + Misses      20340    17781    -2559     
> - Partials     1596     1816     +220     
> ```

This isn't surprising -- coverage can get misattributed when a code path appears to hit a third-part module which is actually a local replacement -- but I was curious where the extra partials came from.

[master](https://app.codecov.io/gh/pulumi/pulumi-terraform-bridge)

| Directory | Lines | Covered | Partial | Missed | % |
| --------- | ----- | ------- | ------- | ------ | - |
| internal | 3003	| 2689	| 11	| 303	| 89.54%
| pf | 6693 |	2743 |	159 |	3791	| 40.98%
| pkg | 39921 | 	23288 |	1391	 | 15242	 | 58.34%
| scripts | 145 |	0	| 0| 	145	|0.00%
| testing | 195	| 74	| 6	|115	| 37.95%
| unstable | 628	| 391 |	19	| 218	| 62.26%
| x | 658	| 122 |	10	| 526	 | 18.54%
| | 51243	| 29307 | 	1596	| 20340 | 57.19%

[PR](https://app.codecov.io/gh/pulumi/pulumi-terraform-bridge/pull/2473/tree?search=&displayType=tree) 

| Directory | Lines | Covered | Partial | Missed | % |
| --------- | ----- | ------- | ------- | ------ | - |
| internal | 3003 | 2689 | 11 | 303 |89.54%
| pf | 6713 | 4031 | 359 | 2323 | 60.05%
| pkg | 39921 | 23984 | 1394 |14543 | 60.08%
| scripts | 145 | 0 | 0 | 145 | 0.00%
| testing |  195 | 74 |6 | 115 | 37.95%
| unstable | 628 | 405 | 19 | 204 | 64.49%
| x | 658 | 483 | 27 | 148 | 73.40%
| | 51263	| 31666 |	1816 |	17781	|61.77%

Essentially we cover a lot more of `pf` and `x` but also picked up some partials. Otherwise things look nearly identical, which is a good sign.

**Comment by VenelinMartinov** [comment]
@blampe thanks for tackling this. I also think we can't really justify the pain this would cause.

Why do you think the approach in https://github.com/pulumi/pulumi-terraform-bridge/pull/2358 of leaving the old stubs around and marking them deprecated might not work? Also, could we try doing that incrementally?

**Comment by blampe** [comment]
> Why do you think the approach in #2358 of leaving the old stubs around and marking them deprecated might not work? Also, could we try doing that incrementally?

@VenelinMartinov I think it can work it just needs to be carefully sequenced. This PR is the first step I'd recommend because it combines everything into the same module. As a next step I would move the relevant code (at least `./pf` and `./x/muxer`, and maybe `./testing` if that's consumed downstream) to their new import paths. Once tests pass on that, I would finally restore the existing modules so imports continue to work.

**Comment by blampe** [comment]
This change is part of the following stack:

- #2473 ◀
    - #2477

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

**Comment by blampe** [comment]
@VenelinMartinov @iwahbe @t0yv0 I put together some additional changes which should cover the whole song and dance.

#2473 #2477 and #2480 break this (very large) change into smaller pieces and should be released together. The net result will be a new source of truth for `pf`, `x/muxer` and `testing`, but existing customers won't be impacted.

I tested #2480 downstream and it appears to work fine (e.g. https://github.com/pulumi/pulumi-azure/pull/2545) modulo some providers which lint against deprecated methods, so I broke that step out into #2482. I just kicked off another downstream test for it -- I'm not sure how to monitor it but the commit is 95a1f1d236ac2b4c5f8c2c35893e59c1751b2c1b.

After those 3 PRs are in we can release https://github.com/pulumi/pulumi-upgrade-provider-action/pull/34 which will automatically fixup these imports for us.

Finally, once that's run its course we can release #2482.

(Let me tell you about a little thing I like to call _monorepo_...)

**Comment by t0yv0** [comment]
I am a big fan of this change. If we can satisfy our internal constraints and add a good bit in release notes for what users have to do, this could lead to a much simpler sane world for everyone. Let me know how I can help (if anything).

**Comment by VenelinMartinov** [comment]
>I just kicked off another downstream test for it -- I'm not sure how to monitor it in aggregate but the commit is https://github.com/pulumi/pulumi-terraform-bridge/commit/95a1f1d236ac2b4c5f8c2c35893e59c1751b2c1b.

We have a script in the bridge repo, which closes all the PRs and checks for failures : `scripts/downstream_checks.py`. I've ran it and there were two network flakes but everything else seems good to go!

Thank you for contributing here, this addresses a persistent source of confusion and errors!

File: pf/tests/integration/program_test.go [hunk 1]
```diff
@@ -24,17 +24,19 @@ import (
...
 	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"sourcegraph.com/sourcegraph/appdash"
 
 	"github.com/pulumi/pulumi-terraform-bridge/v3/unstable/testutil"
 )
 
 func TestBasicProgram(t *testing.T) {
+	if testing.Short() {
```
Comment (VenelinMartinov) [inline pf/tests/integration/program_test.go:37]: I'm not sure we have short and long tests in the bridge - are these needed?

**Review by VenelinMartinov — APPROVED** [review]

**Review by VenelinMartinov — COMMENTED** [review]
Also tried out locally, I believe all tests are ran and errors reported successfully.

File: go.mod [hunk 2]
```diff
@@ -1,31 +1,36 @@
 module github.com/pulumi/pulumi-terraform-bridge/v3
 
-go 1.22
+go 1.22.0
 
-replace github.com/pulumi/pulumi-terraform-bridge/x/muxer => ./x/muxer
+toolchain go1.23.1
```
[the remark this answers is not in the document]
Comment (blampe) [reply go.mod:5]: The minimum go version is still `go 1.22.0`. `toolchain` is only the recommended version for compilation. See the example above.

File: pf/tests/integration/program_test.go (hunk 1 above)
Comment (blampe) [reply pf/tests/integration/program_test.go:37]: I noticed we have different `test.integration` and `test.unit` targets, and `-short` is an easy way to exclude slow integration tests from unit tests (without needing to resort to things like build tags, modules, etc.).

**Review by iwahbe — APPROVED** [review]

**Pull request merged**